src/accelerate/utils/modeling.py

```diff
@@ -407,6 +407,8 @@ def set_module_tensor_to_device(
                 if not is_buffer:
                     module._parameters[tensor_name] = param_cls(new_value, requires_grad=old_value.requires_grad)
         elif isinstance(value, torch.Tensor):
+            if device.type == "xpu" and not is_xpu_available():
+                raise ValueError(f'{device} is not available, you should use device="cpu" instead')
```

Comment:
We should check that device is indeed of type `torch.device` if we use device.type. Can we move this right after if `isinstance(device, int):` block ?

Reply:
Thx for the review! Code updated.